### ISSUE TITLE
Error: (plugins-manager) failed to load 'plugin.py'

### ISSUE THREAD

**Issue by Freekers** [stas-demydiuk/domoticz-plugins-manager#16]
I'm trying to install Plugin Manager, but the plugin does not load. I'm seeing the following errors in my logfile on starting Domoticz:

```
2021-01-21 19:20:49.365  Error: (plugins-manager) failed to load 'plugin.py', Python Path used was '/config/plugins/plugins-manager/:/usr/lib/python38.zip:/usr/lib/python3.8:/usr/lib/python3.8/lib-dynload:/usr/local/lib/python3.8/dist-packages:/usr/lib/python3/dist-packages:/usr/lib/python3.8/dist-packages'.
2021-01-21 19:20:49.365  Error: (Python Plugins Manager) Module Import failed, exception: 'FileNotFoundError'
2021-01-21 19:20:49.365  Error: (Python Plugins Manager) Import detail: File: ./plugins/plugins-manager/plugins.json, Line: -1
```
I'm running Domoticz 2020.2 as Docker using the amd64 image by LinuxServer.io: https://github.com/linuxserver/docker-domoticz . Python is available/installed in the Docker image (Python3.8 + Python3.9) and the Python Path in the error above is valid (exists + has files inside).

I don't get why it says 'FileNotFoundError', as both plugin.py as plugins.json are in the directory:
```
Freekers@Synology:/volume1/docker/domoticz/config/plugins/plugins-manager$ ls
api  CONTRIBUTING.md  frontend  LICENSE  manager.py  plugin.py  plugins.json  plugins.py  __pycache__  README.md
```

I'm already running 2 other plugins without problem (ShellyMQTT and IDetect). What am I missing here?

Thanks!

**Comment by pro2call-nl** [comment]
I have the same problem. No found no solution yet

**Comment by Freekers** [comment]
@stas-demydiuk Do you have any idea perhaps what this might be causing :) ? Thanks!

**Comment by mprins** [comment]
me too using the `linuxserver/domoticz:latest` docker image which has `/config/plugins` as the plugin directory

```
2021-01-31 18:13:06.633 Status: (Plugin manager) Started.
2021-01-31 18:13:06.902 Error: (plugins-manager) failed to load 'plugin.py', Python Path used was '/config/plugins/plugins-manager/:/usr/lib/python38.zip:/usr/lib/python3.8:/usr/lib/python3.8/lib-dynload:/usr/local/lib/python3.8/dist-packages:/usr/lib/python3/dist-packages:/usr/lib/python3.8/dist-packages'.
2021-01-31 18:13:06.902 Error: (Plugin manager) Module Import failed, exception: 'FileNotFoundError'
2021-01-31 18:13:06.902 Error: (Plugin manager) Import detail: File: ./plugins/plugins-manager/plugins.json, Line: -1
2021-01-31 18:13:06.902 Error: (Plugin manager) Error Line details not available.
2021-01-31 18:14:11.964 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
2021-01-31 18:14:25.968 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
2021-01-31 18:14:39.973 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
2021-01-31 18:14:53.977 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
2021-01-31 18:15:08.112 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
2021-01-31 18:15:22.116 Error: Plugin manager hardware (5) thread seems to have ended unexpectedly
```

**Comment by mprins** [comment]
I think `plugins.json` this line: https://github.com/stas-demydiuk/domoticz-plugins-manager/blob/95ec06edd2eaafc344f6c22042b203da1f1ab630/plugins.py#L4
does not resolve to `/config/plugins/plugins-manager/plugins.json`

I see other locations with hardcoded paths that probably won't work either eg. : https://github.com/stas-demydiuk/domoticz-plugins-manager/blob/95ec06edd2eaafc344f6c22042b203da1f1ab630/plugin.py#L60-L61

**Comment by Freekers** [comment]
The templates_path is indeed different for the LinuxServer.io Docker Image. The domoticz-zigbee2mqtt-plugin from stas-demydiuk has a similar issue, which is related to paths as well: https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/556#issuecomment-770223571

**Comment by DavidTre07** [comment]
With the official docker image (domoticz/domoticz), I'm facing the same issue.
Trying to fix it without success: I changed some path but doesn't work... I'm not a developer so hard to find where is the problem.

**Comment by DavidTre07** [comment]
OK I managed the path problem but only for my system:
1- Create a symlink inside the docker : /opt/domoticz/plugins pointing to /opt/domoticz/userdata/plugins
2- In plugin.py replace source_path = os.path.dirname(os.path.abspath(__file__)) + '/frontend' with source_path = '/opt/domoticz/plugins/plugins-manager/frontend'

I'm not sure it will work on domoticz standard installation (outside a docker). This is a workaround as I used absolute path which is definitively not recommended (e.g. if you install domoticz in /home/domoticz it will not work).

**Comment by stas-demydiuk** [comment]
Now should work fine with both setups. I wonder if someone could check

**Comment by DavidTre07** [comment]
OK tested with domoticz official docker image (domoticz/domoticz), works well now.

**Comment by Code2702** [comment]
I still have (this?) problem. I installed PPM on RPi 4b, running Domoticz, eclipse-mosquitto and zigbee2mqtt in Docker Compose.
I have your plugin 'zigbee2mqtt' installed without problem (it's the only plugin at this point). 
My plugins are in /opt/docker/domoticz/plugins.
I cloned https://github.com/ycahome/pp-manager.git in that dir, gave permission and restarted domoticz in docker-compose. 

Then: 
1) I Added PPM in /hardware with plugin 'Idle'. No errors in log yet. 
2) Selected plugin 'Dummy plugin' and click update (not 'Add', so I still have only 1 PPM hardware item). Log says:
2021-03-08 09:40:47.479 Status: (Python Plugin Manager) Stop directive received.
2021-03-08 09:40:47.479 Status: (Python Plugin Manager) Exiting work loop.
2021-03-08 09:40:47.679 Status: (Python Plugin Manager) Stopping threads.
2021-03-08 09:40:47.679 Status: (Python Plugin Manager) Stopped.
2021-03-08 09:40:47.679 Status: (Python Plugin Manager) Started.
**2021-03-08 09:40:47.544 Error: (Python Plugin Manager) Git ErrorNo:2
2021-03-08 09:40:47.544 Error: (Python Plugin Manager) Git StrError:No such file or directory
2021-03-08 09:40:47.566 Error: (Python Plugin Manager) Git ErrorNo:2
2021-03-08 09:40:47.567 Error: (Python Plugin Manager) Git StrError:No such file or directory** 

Seems like a file location problem?

BTW: the permissions/_ownership of the relevant files and directories:
drwxr-xr-x 4 domoticz domoticz 4096 Mar  8 09:37 pp-manager  (dir. /opt/docker/domoticz/plugins/pp-manager)

Files in dir. /opt/docker/domoticz/plugins/pp-manager: 
-rw-r--r-- 1 domoticz domoticz    71 Mar  8 09:36 CONTRIBUTING.md
-rw-r--r-- 1 domoticz domoticz   756 Mar  8 09:36 exceptions.orig
-rw-r--r-- 1 domoticz domoticz 35147 Mar  8 09:36 LICENSE
-rwxr-xr-x 1 domoticz domoticz 41672 Mar  8 09:36 plugin.py
drwxr-xr-x 2 domoticz domoticz  4096 Mar  8 09:37 __pycache__
-rw-r--r-- 1 domoticz domoticz  2194 Mar  8 09:36 README.md
-rw-r--r-- 1 domoticz domoticz   247 Mar  8 09:36 secpoluser.orig
